protocols/dcutr/src/protocol/inbound.rs

```diff
@@ -58,14 +58,23 @@ impl upgrade::InboundUpgrade<NegotiatedSubstream> for Upgrade {
             } else {
                 obs_addrs
                     .into_iter()
-                    .map(Multiaddr::try_from)
+                    .filter_map(|a| match Multiaddr::try_from(a) {
+                        Ok(a) => Some(a),
+                        Err(e) => {
+                            log::debug!("Unable to parse multiaddr: {e}");
+                            None
+                        }
+                    })
                     // Filter out relayed addresses.
-                    .filter(|a| match a {
-                        Ok(a) => !a.iter().any(|p| p == Protocol::P2pCircuit),
-                        Err(_) => true,
+                    .filter(|a| {
+                        if a.iter().any(|p| p == Protocol::P2pCircuit) {
+                            log::debug!("Dropping relayed address {a}");
+                            false
+                        } else {
+                            true
+                        }
                     })
```

Comment:
We could merged these two filter blocks:

```rust 
                    .filter_map(|a| match Multiaddr::try_from(a) {
                        Ok(a) if !a.iter().any(|p| p == Protocol::P2pCircuit) => Some(a),
                        Ok(a) => {
                            log::debug!("Dropping relayed address {a}");
                            None
                        }
                        Err(e) => {
                            log::debug!("Unable to parse multiaddr: {e}");
                            None
                        }
                    })

```
(here and below)
No strong opinion.

Reply:
What would be the benefit of merging the two?

Structure wise, I find the separation to ease understanding.

Reply:
The benefit would be to have a single place where addresses are filtered out. I actually first oversaw that there was a second filter call. 
But that's subjective and nit-picky; feel free to ignore :slightly_smiling_face:.

Reply:
Alternative idea: Keep using `.map(Multiaddr::try_from)` and use `.filter` twice for consistency.

Reply:
> Alternative idea: Keep using `.map(Multiaddr::try_from)` and use `.filter` twice for consistency.

Don't mind me, that doesn't work because we need to unpack the result somewhere.